baseframe/forms/validators.py

```diff
@@ -161,6 +162,30 @@ def compare(self, value, other):
         return value != other
 
 
+class PublicEmailDomain(object):
+    def __init__(self, message=None):
+        self.message = message or _(u'The domain "{domain}" is not a public email domain.')
+
+    def __call__(self, form, field):
+        sniffedmx = mxsniff(field.data)
+        if len(sniffedmx['providers']) > 0:
+            return
+        else:
+            raise StopValidation(self.message.format(domain=field.data))
+
+
+class NotPublicEmailDomain(object):
+    def __init__(self, message=None):
+        self.message = message or _(u'The domain "{domain}" is a public email domain.')
+
+    def __call__(self, form, field):
+        sniffedmx = mxsniff(field.data)
+        if len(sniffedmx['providers']) == 0:
```

Comment:
As above.

Reply:
Can I reuse the above validator inside this one?